pkg/resource/role/delta.go

```diff
@@ -83,6 +83,9 @@ func newResourceDelta(
 			delta.Add("Spec.PermissionsBoundary", a.ko.Spec.PermissionsBoundary, b.ko.Spec.PermissionsBoundary)
 		}
 	}
+	if !reflect.DeepEqual(a.ko.Spec.Policies, b.ko.Spec.Policies) {
+		delta.Add("Spec.Policies", a.ko.Spec.Policies, b.ko.Spec.Policies)
+	}
```

Comment:
Not a big deal but we could use `ackcompare.SliceStringPEqual` instead of deep equal here. Changing the order of policies array will trigger an update reconciliation. IKR the update code path will not call attach/detach policies